bundler/lib/dependabot/bundler/native_helpers.rb

```diff
@@ -1,10 +1,35 @@
 # frozen_string_literal: true
 
+require "bundler"
+require "dependabot/shared_helpers"
+
 module Dependabot
   module Bundler
     module NativeHelpers
-      def self.helper_path
-        "bundle exec ruby #{File.join(native_helpers_root, 'run.rb')}"
+      def self.run_bundler_subprocess(function:, args:, bundler_version:)
+        # Run helper suprocess with all bundler-related ENV variables removed
+        ::Bundler.with_original_env do
```

Comment:
We've flipped this from our original use of `with_clean_env` / `with_unbundled_env` - I don't have extremely strong feelings about flipping this back but it occurs that trying to run Bundler 1 once Dependabot Core itself has installed Bundler 2 `with_original_env` may not be correct, WDYT?

Reply:
Resolving this as it does the right thing and until we introduce bundler 2, we have nothing to be concerned about. We can revisit this in our next step.